### ISSUE TITLE
the test for esnext "RegExp named capture groups" seems weird

### ISSUE THREAD

**Issue by Mensu** [compat-table/compat-table#1180]
From [https://kangax.github.io/compat-table/esnext](https://kangax.github.io/compat-table/esnext) I find the test for esnext "RegExp named capture groups" is like:

```js
var result = /(?<year>\d{4})-(?<month>\d{2})-(?<day>\d{2})/.exec('2016-03-11');
return result.groups.year === '2016'
  && result.groups.month === '03'
  && result.groups.day === '11'
  && result.groups[0] === '2016-03-11'
  && result.groups[1] === '2016'
  && result.groups[2] === '03'
  && result.groups[3] === '11';
```

I am kind of confused comparing it to [the specification proposal](https://github.com/tc39/proposal-regexp-named-groups#high-level-api). Why test ``result.groups[0] === '2016-03-11'`` rather than ``result[0] === '2016-03-11'`` ? Are there anything I am missing, or was this a mistake of the test?

I take it that the case should be like:

```js
var result = /(?<year>\d{4})-(?<month>\d{2})-(?<day>\d{2})/.exec('2016-03-11');
return result.groups.year === '2016'
  && result.groups.month === '03'
  && result.groups.day === '11'
  && result[0] === '2016-03-11'
  && result[1] === '2016'
  && result[2] === '03'
  && result[3] === '11';
```

**Comment by afmenez** [comment]
Yes, and this change makes Chrome 61 pass the test (with the appropriate flag).

**Comment by chicoxyzzy** [comment]
Fixed by #1181
